### DISCUSSION TITLE
Is the RPC client balancing documentation out of date?

### DISCUSSION THREAD

**Discussion by Evgen1000end** [activej/activej#208] [Q&A] [answered]
I'm trying to figure out how to properly set up client balancing for RPC client.

```
@Provides
    io.activej.rpc.client.RpcClient rpcClient(Eventloop eventloop) {
        return io.activej.rpc.client.RpcClient.create(eventloop)
                .withMessageTypes(String.class)
                .withStrategy(server(new InetSocketAddress(HOSTNAME, SERVICE_PORT)));
    }
```

it uses a RpcStrategySingleServer  and RpcClientConnection under the hood.   
The documentation uses completely different classes and it is not clear how to set up the balancing correctly.

Will it be possible to update the documentation with an example of the correct configuration for RpcClient?

Interested in the ability to balance "on the fly" periodically updating the list of addresses

I try this: 

```
@Provides
    io.activej.rpc.client.RpcClient rpcClient(Eventloop eventloop) {
        return io.activej.rpc.client.RpcClient.create(eventloop)
                .withMessageTypes(String.class)
                .withStrategy(roundRobin(servers(new InetSocketAddress(HOSTNAME, SERVICE_PORT), new InetSocketAddress(HOSTNAME, SERVICE_PORT2))));
    }
```

It works perfect but i can't change addresses list dinamically.  I assume that the program will run 24/7 and new RPC servers will be added periodically.

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @Evgen1000end
I have answered on dynamic RPC addresses in #207
